cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -195,6 +212,14 @@ public void initialize(MasterEnvironmentContext context) throws IOException, Api
 
   @Override
   public void destroy() {
+    if (!Strings.isNullOrEmpty(configMapName)) {
+      try {
+        // TODO: CDAP-18504 Create a config map cleaner thread to cleanup in case this deletion fails
+        coreV1Api.deleteNamespacedConfigMap(configMapName, podInfo.getNamespace(), null, null, null, null, null, null);
```

Comment:
ok for now, but did you look into if it's possible to set a parent or something so k8s will delete it automatically?

Reply:
We are planning to use owner reference. When a job is deleted, all the resources associated with that job through owner reference should get deleted. 
Confirmed it through manual testing.